torch/csrc/autograd/variable.cpp

```diff
@@ -87,6 +91,65 @@ std::shared_ptr<Function> VariableImpl::get_grad_accumulator() {
   return result;
 }
 
+VariableViewImpl::VariableViewImpl(Variable base_, at::Tensor data_)
+  : VariableImpl(std::move(data_))
+  , base(std::move(base_))
+  , attr_version(0) {
+  if (!base.defined()) {
+    throw std::runtime_error("base is undefined");
+  }
+  if (base.is_view()) {
+    base = base.base();
+  }
+  is_view = true;
+  version_counter = base.version_counter();
+  attr_version = version_counter.current_version();
```

Comment:
I think it would be safer to set `attr_version` to an invalid value like -1. You didn't set up a `grad_fn` in here, but it might be necessary, so we better enter the slow path of `get_grad_fn()` next time

Reply:
I've changed the constructor so it takes a grad_fn